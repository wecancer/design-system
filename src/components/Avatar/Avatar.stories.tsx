import React from 'react'
import {Story} from '@storybook/react'

import Avatar, {Props} from '.'

const Template: Story<Props> = (args) => <Avatar {...args} />

export const Default = Template.bind({})

Default.args = {
  size: 48,
  src: 'https://randomuser.me/api/portraits/women/42.jpg',
  initials: 'ANA',
}

export default {
  title: 'Components/Avatar',
  component: Avatar,
  argTypes: {
    status: {
      control: {
        type: 'select',
        options: ['online', 'busy', 'away'],
      },
    },
  },
}
