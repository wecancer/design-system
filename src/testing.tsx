import React from 'react'
import {render, RenderResult} from '@testing-library/react'
import WecancerProvider from './WecancerProvider'

type Props = {
  children: React.ReactNode
}

const AllTheProviders = ({children}: Props) => {
  return <WecancerProvider>{children}</WecancerProvider>
}

// eslint-disable-next-line
const appRender = (ui: React.ReactElement, options?: any): RenderResult =>
  render(ui, {wrapper: AllTheProviders, ...options})

// re-export everything
export * from '@testing-library/react'

// override render method
export {appRender as render}
