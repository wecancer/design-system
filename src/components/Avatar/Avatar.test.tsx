import React from 'react'

import Avatar from '.'
import {render} from '../../testing'

describe('<Avatar />', () => {
  it('should render snapshot correctly', () => {
    const {container} = render(
      <>
        <Avatar src="https://randomuser.me/api/portraits/women/42.jpg" />
        <Avatar initials="ANA" />
        <Avatar initials="ANA" size={100} />
        <Avatar initials="ANA" status="away" />
        <Avatar initials="ANA" status="busy" />
        <Avatar initials="ANA" status="online" />
        <Avatar initials="ANA" src="https://randomuser.me/api/portraits/women/42.jpg" />
        <Avatar src="https://randomuser.me/api/portraits/women/42.jpg" status="away" />
        <Avatar src="https://randomuser.me/api/portraits/women/42.jpg" status="busy" />
        <Avatar src="https://randomuser.me/api/portraits/women/42.jpg" status="online" />
      </>,
    )
    expect(container.firstChild).toMatchSnapshot()
  })

  it('should have the image tag with src property', () => {
    const {container} = render(<Avatar src="https://randomuser.me/api/portraits/women/42.jpg" />)
    expect(container.querySelector('img')).toHaveProperty('src')
  })

  it('must not have an image tag when src is not defined', () => {
    const {container} = render(<Avatar initials="AB" />)
    expect(container.querySelector('img')).toBeNull()
  })
})
