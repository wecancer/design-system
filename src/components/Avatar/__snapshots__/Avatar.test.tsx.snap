// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Avatar /> should render snapshot correctly 1`] = `
<div
  class="sc-bdvvtL iMtUhn"
>
  <figure>
    <img
      alt=""
      src="https://randomuser.me/api/portraits/women/42.jpg"
    />
  </figure>
</div>
`;
